### PR TITLE
[RDY] Update chapter 2

### PR DESCRIPTION
**Pull request by Hjdskes**
@Balletie Can you go over the Stratego/DynSem sections and see if those need updating after DynSem's rewrite?

Ignore the wrong branch name :P

### CODE REVIEW

File: final/background/repl.tex [hunk 1]
```diff
@@ -155,9 +154,10 @@ \subsection{Functionality}
 \paragraph{Redefining identifiers} When using a REPL in an exploratory manner,
 it is not uncommon to want to redefine an identifier's type or to completely
 reimplement a method. In this way, a REPL can be different than its host
-language, especially if the host language is a functional language that doesn't
-allow variable's values to change once initiated.
+language, especially if the host language is a functional language that does not
+allow variables' values to change once initiated.
 
+% FIXME: shows a requirement between the lines.
```
Comment (Hjdskes) [inline final/background/repl.tex]: I still have to resolve this one. Suggestions, maybe?

File: final/background/spoofax.tex [hunk 2]
```diff
@@ -27,33 +27,34 @@ \section{The Spoofax Language Workbench}
...
 representation of a program to an abstract syntax tree (AST)
 representation. In Spoofax, the syntax is declared with a domain
 specific language (DSL) called SDF.
-\item \hyperref[sec:static-analysis]{Static Semantics}: The AST then goes through static analysis (type
+\item \hyperref[ssec:static-analysis]{Static Semantics}: The AST then goes through static analysis (type
 checking, name binding and variable scoping), to test if the
 program is well-formed. Static semantics describe the rules for the
 static analysis step. Spoofax provides two DSLs for the
-specification of this step: the TS type specification language and
+specification of this step: the TS Type Specification language and
```
Comment (gfokkema) [inline final/background/spoofax.tex]: Reading this it seems to me like the user has a choice on whether he uses NABL or TS for static analysis, but these languages are not mutually exclusive and each have their own task (as is properly explained further on).

Comment (Hjdskes) [inline final/background/spoofax.tex]: I can add a few words to remove this possible confusion, however I think for an introduction it is fine as-is: removing the confusion by explaining that one can 1) use only NaBL, 2) use only TS, or 3) use both, makes the introduction too long.

@Balletie thougts?

File: final/background.tex [hunk 3]
```diff
@@ -1,9 +1,18 @@
 \chapter{Background}
 \label{cha:background}
 
+This chapter gives the required background information required to
```
Comment (Balletie) [inline final/background.tex]: You say required twice here.

File: final/background.tex [hunk 4]
```diff
@@ -1,9 +1,18 @@
...
+understand the rest of this report. If the reader is already well-versed in
+Spoofax and familiar with a REPL, this chapter can be skipped.
```
Comment (Balletie) [inline final/background.tex]: Maybe "with the concept of a REPL"

File: final/background/repl.tex (hunk 1 above)
Comment (Balletie) [inline final/background/repl.tex]: > Typically a REPL for a language has the same error reporting as its interpreter or compiler as if it was invoked with the command line interface. **[List examples for general claim]**. A REPL embedded within an IDE could additionally show errors while the user is still typing.

File: final/background/spoofax.tex (hunk 2 above)
Comment (Balletie) [inline final/background/spoofax.tex]: > Spoofax provides two DSLs that can specify the two distinct parts of this step: ...

**Pull request merged**